### ISSUE TITLE
Protected member access allowed in external-constraints from class

### ISSUE THREAD

**Issue by jrudess** [MikePopoloski/slang#1109]
The following example analyzes without error, but since 'a' is a protected member in 'C', I believe there should be a protected member access error in the ```randomize() with``` constraints.

```Systemverilog
class C;
    protected int a;
    rand bit b;
endclass

class T;
    function f();
        C c = new();
        c.randomize() with {
            if (a == 3) {
                b == 1'b1;
            }
        };
    endfunction
endclass
```

**Comment by MikePopoloski** [comment]
Hmm, the LRM does not say one way or the other. VCS and Questa error, Xcelium allows it. It makes sense to me to disallow it.

**Comment by jrudess** [comment]
VCS used to allow it in older versions as well.  It was something they started to enforce sometime after the 2020 versions I think.

**Comment by MikePopoloski** [comment]
Fixed in 2ba4087145919a58ba1221577403c134e7b48e58
